UI/MeterBrowseWnd.cpp

```diff
@@ -358,28 +358,43 @@ void MeterBrowseWnd::UpdateEffectLabelsAndValues(GG::Y& top) {
             if (const auto& building = std::dynamic_pointer_cast<const Building>(source))
                 if (const auto& planet = Objects().get<Planet>(building->PlanetID()))
                     name = planet->Name();
-            // same effect from several planets?
+            // Some effects are triggered by every building in the own empire.
+            // To avoid excessive effect list, we combine same effects.
+            // Local (same planet or system) buildings have a stronger effect
+            // and are kept apart.
+            // So this changes e.g.
+            // Likes <own planet> building Interstellar Lighthouse +4.00
+            //    Likes <other 1> building Interstellar Lighthouse +0.71
+            //    Likes <other 2> building Interstellar Lighthouse +0.71
+            // to
+            // Likes <own planet> building Interstellar Lighthouse +4.00
+            //          Likes 2 x building Interstellar Lighthouse +1.41
+            // Note that if there is more than one of the building on other
```

Comment:
meaning of this comment is unclear... is this an issue specific to how the loking / disliking effects are scripted, or a more fundamental issue to how effect accounting entries are combined?

Reply:
I guess before I upload more changes, let's discuss what the outcome should be. So far the code produces one line per effect strength, so e.g. you'll get:
- Likes Orion I building Interstellar Lighthouse +4.00 // own planet -> strongest effect
- Likes 2 x building Interstellar Lighthouse +2.0 // Orion II and III, same system 1 per building
- Likes 2 x building Interstellar Lighthouse +1.41 // Other planets, 1 / sqrt(num) per planet

The code knows nothing of effect groups so you may also get:
- Likes Orion I building Interstellar Lighthouse +4.00 // own planet -> strongest effect
- Likes 2 x building Interstellar Lighthouse +2.0 // Orion II and Antares I (a single remote also gives 1.0)

If the list can be unfolded (note that the list is displayed via mouse-over, so this may be tricky), I could just as well combine all effects of one building type into one line and collect the values together with the planet names to be shown when the list is unfolded.

If unfolding is not done, I think keeping separate lines makes the list easier to understand. The code could then be simplified by only counting the number of planets, thus replacing combined_names and combined_meter_change by a single integer.

Reply:
Separate lines for separate effect sizes is good / fine I think.

My only concern is whether 3 effects that each give +2 for a total of +6 should be shown as
`Likes 3x building Interstellar Lighthouse +2.0` or `Likes 3x building Interstellar Lighthouse +6.0` where the +YY number is the total, not the per-effect contribution. I lean toward the latter, but not strongly.

Reply:
Well, so far you could always add up the numbers shown in on the right column to get the target effect. This will no longer work and when we combine several lines, but do not add their effects strengths. That's why I'd rather show the combined effect

Reply:
Of course with a little more display hacking, we could also make it something like that:
Likes 3x building Interstellar Lighthouse 3 x +2.0 = 6.0

Reply:
I think we're saying the same thing...?

Reply:
I hope the comments are better now. I also simplified the code by removing combined_meter_change. When combining only identical effect, multiplication does the job.
I also added a small script fix. Looking at the numbers it occurred to me that sqrt(5) is not 2.04!